apache/incubator-zeppelin
# Zeppelin-540: Notebook versioning control, commit from frontend notebook menu

### What is this PR for?
This PR improves the front-end experience to use git versioning on notebooks. 
Any community feedback is welcome.

### What type of PR is it?
Improvement

### Todos
* [x] - front-end and back-end changes
* [x]  - add `checkpoint` interface
* [x] - ~~error propagation to front-end~~ in a separate PR
* [x] - UI feedback?

### Is there a relevant Jira issue?
[ZEPPELIN-540](https://issues.apache.org/jira/browse/ZEPPELIN-540)

### How should this be tested?
1. Uncomment `GitNotebookRepo` as a storage in `/conf/zeppelin-site.xml`
2. Start Zeppelin
3. Make changes to some notebook
4. Use `git` menu to commit
5. To check whether successfully committed, do `git log` in `/notebooks` folder

### Screenshots (if appropriate)
![zeppelin commit notebook](https://cloud.githubusercontent.com/assets/1642088/12534987/3a0e72c2-c2b5-11e5-827e-33ba36826637.gif)


### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No
